flows/deploy.py

```diff
@@ -0,0 +1,58 @@
+import os
+import subprocess
+import sys
+
+from packaging.version import Version
+
+import prefect
+
+# The version oldest version this test runs with
+SUPPORTED_VERSION = "2.6.0"
+
+
+if Version(prefect.__version__) < Version(SUPPORTED_VERSION):
+    raise NotImplementedError()
+
+
+async def main():
+    if Version(prefect.__version__) >= Version("2.19.0"):
+        try:
+            TEST_SERVER_VERSION = os.environ.get(
+                "TEST_SERVER_VERSION", prefect.__version__
+            )
+            # Work pool became GA in 2.8.0
+            if Version(prefect.__version__) >= Version("2.8") and Version(
+                TEST_SERVER_VERSION
+            ) >= Version("2.8"):
+                subprocess.check_call(
+                    ["prefect", "work-pool", "create", "test-pool", "-t", "process"],
+                    stdout=sys.stdout,
+                    stderr=sys.stderr,
+                )
+
+                flow_instance = await prefect.flow.from_source(
+                    source="https://github.com/PrefectHQ/prefect-recipes.git",
+                    entrypoint="flows-starter/hello.py:hello",
+                )
+
+                await flow_instance.deploy(
+                    name="demo-deployment",
+                    work_pool_name="test-pool",
+                )
+
+                flow_run_state = flow_instance._run("demo-deployment")
```

Comment:
I think it'd make more sense to use `run_deployment` followed by a call to `prefect flow-run execute` to smoke test an end-to-end run.

Reply:
That makes sense, I've applied in [2a6086d](https://github.com/PrefectHQ/prefect/pull/13392/commits/2a6086d38b82e03fb7739833f94351a78d38593b)